File: android/src/main/kotlin/com/retozu/flutter_amazon_chime/ActiveSpeakerObserver.kt
package com.retozu.flutter_amazon_chime

import com.amazonaws.services.chime.sdk.meetings.audiovideo.AttendeeInfo
import com.amazonaws.services.chime.sdk.meetings.audiovideo.audio.activespeakerdetector.ActiveSpeakerObserver

class ChimeActiveSpeakerObserver(private val chimeFlutterApi: ChimeFlutterApi) : ActiveSpeakerObserver {
    override val scoreCallbackIntervalMs: Int? = null

    override fun onActiveSpeakerDetected(attendeeInfo: Array<AttendeeInfo>) {
        val ids = attendeeInfo.map { it.attendeeId as String? }
        chimeFlutterApi.onActiveSpeakersChanged(ids) {}
    }

    override fun onActiveSpeakerScoreChanged(scores: Map<AttendeeInfo, Double>) {
        // not forwarded
    }
}

File: ios/Classes/ActiveSpeakerObserver.swift
import AmazonChimeSDK

class ChimeActiveSpeakerObserver: ActiveSpeakerObserver {
    let chimeFlutterApi: ChimeFlutterApi

    var observerId: String = UUID().uuidString

    init(chimeFlutterApi: ChimeFlutterApi) {
        self.chimeFlutterApi = chimeFlutterApi
    }

    func activeSpeakerDidDetect(attendeeInfo: [AttendeeInfo]) {
        let ids = attendeeInfo.map { $0.attendeeId as String? }
        chimeFlutterApi.onActiveSpeakersChanged(attendeeIds: ids) { _ in }
    }
}
